src/Orchard/Data/Migration/AutomaticDataMigrations.cs

```diff
@@ -58,5 +68,15 @@ public class AutomaticDataMigrations : IOrchardShellEvents {
         public void Terminating() {
             // No-op.
         }
+
+        /// <summary>
+        /// This ensures that the framework migrations have run for the distributed locking feature, as existing Orchard installations will not have the required tables when upgrading.
+        /// </summary>
+        private void EnsureDistributedLockSchema() {
+            // Ensure the distributed lock record schema exists.
+            var schemaBuilder = new SchemaBuilder(_dataMigrationInterpreter);
+            var distributedLockSchemaBuilder = new DistributedLockSchemaBuilder(_shellSettings, schemaBuilder);
+            distributedLockSchemaBuilder.EnsureSchema();

```

Comment:
This code would then call a method on that `CoreSchemaBuilder` I suggested in my earlier comment.